spec/actions/pkg_spec.cr

```diff
@@ -165,10 +165,10 @@ module DppmRestApi::Actions::Pkg
         BuildResponse.from_json(response.body).should_be_successful_for "dppm"
       end
     end
-    it "responds with Internal Server Error when given an invalid package name" do
+    it "responds with Bad Request when given an invalid package name" do
```

Comment:
This spec is failing due to an issue in the library I think.

Reply:
The dppm library? I don't think so, it doesn't know the concept of API routes. I figured out why.

Reply:
The dppm library doesn't need to know API routes to raise an exception upon receiving an invalid package name. Which is what this spec is testing. Sorry, I was trying to point out the body, not which HTTP status was raised..

Reply:
```crystal
post fmt_route "/$(echo all your base are belong to us!)/build"
# (which does)
pkg = pfx.new_pkg "$(echo all your base are belong to us!)", version: nil
# this fails both because there's no underscore in the package name
# with no version specified, and because the package name has invalid
# characters
init_done = false
pkg.build confirmation: false do
  # this block should never get yielded to, due to input validations raising
  # an exception before the block is called
  init_done = true
rescue e
  # which means this exception is raised.
  raise InternalServerError.new context, cause: e if init_done
  raise BadRequest.new context, cause: e
end
# making this line fail
response.status_code.should eq HTTP::Status::BAD_REQUEST.value
```

Reply:
This is due to some methods not being catched in `Actions::Pkg` (see my code suggestion).
A package can be built internally from a source (`Src` object) in order to take the latest version. If it doesn't exist, an exception is raised.
Setting the package name + version doesn't require to read a source package.